### PR TITLE
Kernel+Userland: Basic process locking for ioctl operations on DisplayConnectors

### PR DESCRIPTION
**Pull request by supercomputer7**
Now that the infrastructure of the Graphics subsystem is quite stable, I'm trying to fix a long-standing problem, which is the lack of locking on display connector devices. Reading and writing to the framebuffer controlled by the display connector is not a huge problem, for now (it might be a problem if other program tries to interfere with WindowServer but that could be solved with POSIX locking). The real problem is some program that will try to do ioctl operations on a display connector without the WindowServer being aware of that which can lead to very bad situations (e.g. assuming that one framebuffer is running at a known resolution but it is not but we don't know about it).

To solve this, I invented a "master" ioctl to take "ownership" and release that ownership of a display connector device. To ensure we will not hold a Process object forever just because it has an ownership over a display connector, it must be a WeakPtr then, and if the process is gone, someone else can take ownership.

### CODE REVIEW

File: Userland/Utilities/try-override-graphics.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*
```
Comment (FireFox317) [inline Userland/Utilities/try-override-graphics.cpp:1]: This seems to be some testing code that got committed by accident?

Comment (supercomputer7) [reply Userland/Utilities/try-override-graphics.cpp:1]: Not really, I wanted to put that as a test, but CI doesn't run in GUI mode, so I'm not sure what to do with this small utility.

Comment (supercomputer7) [reply Userland/Utilities/try-override-graphics.cpp:1]: It seems like there's no good place for this test program to reside in, so I'll remove it and mark this resolved :)

**Comment by circl-lastname** [comment]
There are build errors on CI

**Comment by supercomputer7** [comment]
This is not stale.

**Comment by linusg** [comment]
This has a conflict unfortunately

File: Userland/Libraries/LibEDID/EDID.cpp [hunk 2]
```diff
@@ -201,7 +201,7 @@ ErrorOr<Parser> Parser::from_bytes(ByteBuffer&& bytes)
 }
 
 #ifndef KERNEL
-ErrorOr<Parser> Parser::from_framebuffer_device(int framebuffer_fd, size_t head)
+ErrorOr<Parser> Parser::from_displayconnector_device(int framebuffer_fd, size_t head)
```
Comment (linusg) [inline Userland/Libraries/LibEDID/EDID.cpp:204]: - `display_connector` to match `DisplayConnector`
- `framebuffer_fd`?

**Review by linusg — CHANGES_REQUESTED** [review]

**Pull request merged**